sync/task_pool.go

```diff
@@ -249,10 +249,20 @@ func (p *TaskPool) IsClosed() bool {
 	}
 }
 
+// Close close, terminate task
 func (p *TaskPool) Close() {
 	p.stop()
 	p.wg.Wait()
 	for i := range p.qArray {
 		close(p.qArray[i])
 	}
 }
+
+// CloseTillTaskComplete close till task complete
+func (p *TaskPool) CloseTillTaskComplete() {
```

Comment:
lack of the uint test. also i think, if only call the CloseTillTaskComplete this maybe always wait at p.wg.Wait(), the program can't exit